### ISSUE TITLE
Feature request: allow aws_ecs_task_definitions to track the latest revision

### ISSUE THREAD

**Issue by dennari** [hashicorp/terraform#8449]
I'm using the AWS Elastic Container Service managed wholly with Terraform. Deploying an `aws_ecs_service` with a new Docker image means tainting the associated `aws_ecs_task_definition` and this works great.

An example configuration could look like this:

```
resource "template_file" "foo" {
  template = "${file("${path.module}/taskDefinitions/foo.json")}"
  vars {
    image_tag = "latest"
  }
  lifecycle {
    ignore_changes = ["vars.image_tag"]
  }
}

resource "aws_ecs_task_definition" "foo" {
  family                = "foo"
  container_definitions = "${template_file.foo.rendered}"
}

resource "aws_ecs_service" "foo" {
  name            = "foo"
  cluster         = "${aws_ecs_cluster.foo.id}"
  task_definition = "${aws_ecs_task_definition.foo.family}:${aws_ecs_task_definition.foo.revision}"
  desired_count   = 2
}
```

Problems arise when I'd like to restart the service using the AWS API's directly as part of a CI process. The only thing that changes in the states of the associated task definition and service, is the revision number of the task definition (and possibly the image, but this could be handled with `ignore_changes`). Making an external deployment then results in the following plan:

```
~ aws_ecs_service.foo
    task_definition: "foo:2" => "foo:1"
```

i.e., Terraform would like to roll the service back to itse previous state.

It would be great if `aws_ecs_task_definition` could optionally update its revision to the latest one during the refresh phase iff the only diff is the revision and the refreshed revision is higher.

If there are other ways to achieve Terraform being in sync with the revisions I'd be very interested in hearing them.

**Comment by dennari** [comment]
I think another, and more general, way to achieve the described goal would be to also have `data aws_ecs_task_definition` and a `max()` interpolation function. Then I could do:

```
resource "aws_ecs_service" "foo" {
  name            = "foo"
  cluster         = "${aws_ecs_cluster.foo.id}"
  task_definition = "${aws_ecs_task_definition.foo.family}:${max("${aws_ecs_task_definition.foo.revision}","${data.aws_ecs_task_definition.foo.revision}")}"
  desired_count   = 2
}
```

**Comment by RyanBowlby-Reflektion** [comment]
https://github.com/hashicorp/terraform/pull/7230

See the above new data resource. I'm not entirely convinced this is the right solution though. I think using the data resource above only works if the task definition already exists. What is to be done about the very first terraform apply?

**Comment by dennari** [comment]
@RyanBowlby-Reflektion, good point. There would need to be logic that'd recognize that both the resource and the datasource are defined and in this case default the datasource's revision to 0 on the initial apply. I agree this sounds a bit complicated and wouldn't allow the datasource to have pretty much any other attributes than the revision, since the other ones probably wouldn't have any acceptable 'initial' value.

That said, for the described use case no other attributes are needed.

**Comment by dennari** [comment]
PR #8509 looks great. However it seems to require applying in steps, first creating the task definition resources and then adding the datasources. Please correct me if I'm wrong.

I realize there's also an issue with the image tags. Datasources don't support the ignore_changes lifecycle attribute, so in the described use case the deprecated `template_file.foo` could not be switched into a datasource.

**Comment by stack72** [comment]
Closed via #8509 - this is now a data source that allows you to get the latest version
